src/component/mxgraph/MxGraphConfigurator.ts

```diff
@@ -91,17 +91,66 @@ export default class MxGraphConfigurator {
       // only the ctrl key or the meta key on mac
       const isZoomWheelEvent = (evt.ctrlKey || (mxClient.IS_MAC && evt.metaKey)) && !evt.altKey && !evt.shiftKey;
       if (isZoomWheelEvent) {
-        this.zoom(up);
+        this.graph.performZoom(up, evt);
         mxEvent.consume(evt);
       }
     }, this.container);
   }
+}
+
+class BpmnMxGraph extends mxGraph {
+  private cumulativeZoomFactor = 1;
+
+  constructor(readonly container: HTMLElement) {
+    super(container);
+  }
+
+  // override fit to set initial cumulativeZoomFactor
+  fit(order: number, keepOrigin?: boolean, margin?: number, enabled?: boolean, ignoreWidth?: boolean, ignoreHeight?: boolean, maxHeight?: number): number {
+    const scale = super.fit(order, keepOrigin, margin, enabled, ignoreWidth, ignoreHeight, maxHeight);
+    this.cumulativeZoomFactor = scale;
+    return scale;
+  }
 
-  private zoom(zoomIn: boolean): void {
-    if (zoomIn) {
-      this.graph.zoomIn();
+  // solution inspired by https://github.com/algenty/grafana-flowcharting/blob/0.9.0/src/graph_class.ts#L1254
+  public performZoom(up: boolean, evt: MouseEvent): void {
+    const rect = this.container.getBoundingClientRect();
+    const x = evt.clientX - rect.left;
+    const y = evt.clientY - rect.top;
+    this.zoomToMousePointer(up, x, y);
+  }
+
+  private zoomToMousePointer(up: boolean, offsetX: number, offsetY: number): void {
```

Comment:
Why do you not use [zoomTo](https://jgraph.github.io/mxgraph/docs/js-api/files/view/mxGraph-js.html#mxGraph.zoomTo) ?

Reply:
Agree, there is no ref to mouse here, we pass coordinates